src/hooks/useCurrentBlockTimestamp.ts

```diff
@@ -1,10 +1,20 @@
 import { BigNumber } from '@ethersproject/bignumber'
 import { useSingleCallResult } from 'lib/hooks/multicall'
+import { useEffect, useState } from 'react'
 
 import { useInterfaceMulticall } from './useContract'
 
 // gets the current timestamp from the blockchain
 export default function useCurrentBlockTimestamp(): BigNumber | undefined {
+  const [lastBlock, setLastBlock] = useState<BigNumber | undefined>()
```

Comment:
can you simplify with `usePreviousValue`?

Reply:
`usePrevious` will not work because it is still based on referential equality.